spec.bs

```diff
@@ -6,57 +6,417 @@ Status: CG-DRAFT
 
-  Promise<any> append(DOMString key, DOMString value);
+The {{SharedStorageWorklet}}'s worklet global scope type is {{SharedStorageWorkletGlobalScope}}. The scripts should define and {{register}} one or more {{SharedStorageOperation}}s. 
 
-  Promise<any> delete(DOMString key);
+<xmp class='idl'>
+  [Exposed=SharedStorageWorklet, Global=SharedStorageWorklet]
+  interface SharedStorageWorkletGlobalScope : WorkletGlobalScope {
+    undefined register(DOMString name,
+                       SharedStorageOperationConstructor operationCtor);
+  };
+
+  callback SharedStorageOperationConstructor = SharedStorageOperation(object options);
+
+  [Exposed=SharedStorageWorklet]
+  interface SharedStorageOperation {
+  };
+</xmp>
+
+Currently each {{SharedStorageOperation}} registered via {{SharedStorageWorkletGlobalScope/register()}} must be one of the following two types:
+* {{SharedStorageRunOperation}}
+* {{SharedStorageSelectURLOperation}}
+
+A {{SharedStorageRunOperation}} is an {{SharedStorageOperation}} that has no JavaScript return value, and hence is especially suited for calling into the private aggregation service.
+
+A {{SharedStorageSelectURLOperation}} is an {{SharedStorageOperation}} that takes in a list of {{SharedStorageUrlWithMetadata}} dictionaries (i.e. URLs each wrapped with metadata) and returns a {{long}} integer index specifying which of these URLs should be selected.
+
+<xmp class='idl'>
+  [Exposed=SharedStorageWorklet]
+  interface SharedStorageRunOperation : SharedStorageOperation {
+    Promise<undefined> run(object data);
+  };
 
-  Promise<any> clear();
+  [Exposed=SharedStorageWorklet]
+  interface SharedStorageSelectURLOperation : SharedStorageOperation {
+    Promise<long> run(object data,
+                      FrozenArray<SharedStorageUrlWithMetadata> urls);
+  };
 
-  Promise<USVString> selectURL(DOMString name,
+  dictionary SharedStorageUrlWithMetadata {
+    USVString url;
+    object reportingMetadata;
+  };
+</xmp>
+
+  ## Registering a Worklet ## {#register-worklet}
+
+  <div algorithm>
+    The <dfn method for="SharedStorageWorkletGlobalScope">register(|name|, |operationCtor|)</dfn> method steps are:
+
+    1. If |name| is missing or empty, throw a {{TypeError}}.
+    1. Let |operation_map| be the internal operation <a href="https://infra.spec.whatwg.org/#ordered-map">map</a> for this scope.
+    1. If |name| already exists as a key in |operation_map|, throw a {{TypeError}}.
+    1. If |operationCtor| is missing or not a constructor, throw a {{TypeError}}.
+    1. Let |prototype| be the class prototype for |operationCtor|.
+    1. If |prototype| is not an object, throw a {{TypeError}}.
+    1. Let |run| be the "run" method in |prototype|.
+    1. If |run| is missing or not a function, throw a {{TypeError}}.
+    1. <a href="https://infra.spec.whatwg.org/#map-set">Set</a> the value of |name| to |run| in |operation_map|.
```

Comment:
You can use `[=map/set=]` instead of writing the link yourself.

Reply:
Done.